download-utils/src/lib.rs

```diff
@@ -108,17 +110,79 @@ pub fn download_file(url: &str, destination_file: &Path) -> Result<(), String> {
     Ok(())
 }
 
-fn extract_archive(archive_filename: &Path, destination_dir: &Path) -> Result<(), String> {
+const MAXIMUM_GENESIS_ARCHIVE_SIZE: u64 = 10 * 1024 * 1024; // 10 MiB
+
+fn is_valid_archive_entry(parts: &[&str], kind: tar::EntryType) -> bool {
+    use tar::EntryType::{Directory, GNUSparse, Regular};
+
+    trace!("validating: {:?} {:?}", parts, kind);
+    match (parts, kind) {
+        (["genesis.bin"], Regular) => true,
+        (["rocksdb"], Directory) => true,
+        (["rocksdb", ..], GNUSparse) => true,
```

Comment:
Hello, shiny subslice pattern! :)

Previously:
```
error[E0658]: subslice patterns are unstable
  --> <source>:40:22
   |
40 |         (["rocksdb", ..], GNUSparse) => true,
   |                      ^^
   |
   = note: for more information, see https://github.com/rust-lang/rust/issues/62254
```

Reply:
heh cool!

Reply:
Does it match anything after the rocksdb though? Could I have `rocksdb/../../../malicious_file`?

Reply:
@sakridge Nice catch! Fixed...